cg/models/report/validators.py

```diff
@@ -1,100 +1,77 @@
 import logging
 from datetime import datetime
 from pathlib import Path
-from typing import Union
+from typing import Any, Optional
+
+from pydantic import ValidationInfo
 
 from cg.constants import (
     BALSAMIC_ANALYSIS_TYPE,
     NA_FIELD,
     NO_FIELD,
     PRECISION,
     REPORT_GENDER,
-    REPORT_SUPPORTED_PIPELINES,
     YES_FIELD,
-    Pipeline,
 )
+from cg.constants.constants import Pipeline, PrepCategory
+from cg.constants.subject import Gender
 from cg.models.orders.constants import OrderType
 
 LOG = logging.getLogger(__name__)
 
 
-def validate_empty_field(value: Union[int, str]) -> str:
-    """Formats an empty value to be included in the report as N/A."""
+def get_report_string(value: Any) -> str:
+    """Return report adapted string."""
     return str(value) if value else NA_FIELD
 
 
-def validate_boolean(value: Union[bool, str]) -> str:
-    """Formats a boolean value for the delivery report."""
-    if isinstance(value, bool) or value:
-        if str(value) == "True":
-            return YES_FIELD
-        if str(value) == "False":
-            return NO_FIELD
+def get_boolean_as_string(value: Optional[bool]) -> str:
+    """Return delivery report adapted string representation of a boolean."""
+    if isinstance(value, bool):
+        return YES_FIELD if value else NO_FIELD
     return NA_FIELD
 
 
-def validate_float(value: Union[float, str]) -> str:
-    """Returns a processed float value."""
+def get_float_as_string(value: Optional[float]) -> str:
+    """Return string representation of a float value."""
     return str(round(float(value), PRECISION)) if value or isinstance(value, float) else NA_FIELD
 
 
-def validate_percentage(value: Union[float, str]) -> str:
-    """Returns a processed float value as a percentage."""
-    return validate_float(float(value) * 100) if value else NA_FIELD
+def get_float_as_percentage(value: Optional[float]) -> str:
+    """Return string percentage representation of a float value."""
+    return get_float_as_string(value * 100) if value or isinstance(value, float) else NA_FIELD
```

Comment:
What is the intention with "or isinstance(value, float)"? Given lazy loading, I would guess that the only value passed to "isinstance(value,float)" is when value == None?

Reply:
Just to take into consideration the possibility of zero input values. It's unlikely that we'll get them, as it would mean ignoring some QC checks, but I suppose it's a good practice to account for the majority of possible scenarios